File: YouMedServer/Models/Entities/Clinic.cs
using System.ComponentModel.DataAnnotations;
namespace YouMedServer.Models.Entities
{
    public class Clinic
    {
        [Key]
        public int ClinicID { get; set; }

        [Required]
        [MaxLength(255)]
        public required string Name { get; set; }

        [MaxLength(255)]
        public string? ClinicAddress { get; set; }

        public string? Introduction { get; set; }

        [MaxLength(20)]
        public string? PhoneNumber { get; set; }

        public DateTime CreatedAt { get; set; }
    }
}

File: YouMedApp/types/Clinic.ts
import Specialties from "./Specialties";
import WorkingHours from "./WorkingHours";

export default interface Clinic {
    clinicID: number;
    name: string;
    clinicAddress: string;
    introduction: string;
    phoneNumber: string;
    createdAt: string; 
    specialties: Specialties[]
    clinicWorkingHours: WorkingHours[]
    latitude: number; 
    longitude: number; 
    rating?: number;
    distance?: string;
    openingHours?: string;
  }